docs/topics/patterns/instant-messaging.mdx

```diff
@@ -0,0 +1,355 @@
+
+## Components
+
+Instant messaging is separated into two parts:
+
+* Message storage
+* Delivery
+
+These core components allow users to join your chat at any time and view the message history and enable them to receive new messages in real time.
+
+### Message storage
+
+Most instant messaging platforms store messages to allow users to view the conversation history if messages were sent prior to their arrival. Part of the startup tasks when entering a chat room is loading said history and displaying it on screen. This can easily be done with Momento in two ways: with a **sorted set** or a **list** cache item, each with their own tradeoffs.
+
+#### Sorted sets
+
+Typically used for things like [leaderboards](https://www.gomomento.com/blog/build-on-momento-tips-and-tricks-for-a-lightning-fast-game-leaderboard), a [sorted set](../../cache/develop/basics/datatypes#sorted-sets) is a collection data type that stores multiple value/score elements. In each element, the value is a `string` and the score is an `integer`. Specifically with instant messaging, you can store the time the message was sent as the *score* and a stringified JSON object as the *value*. Storing the message timestamp as the score will automatically allow you to sort messages by time since sorted sets do that by default.
+
+:::tip
+To store the time as the score, you must use an integer representation of the time. Store either ticks or time in milliseconds since the Unix epoch.
+:::
+
+Sorted sets have a [rich set of features](../../cache/develop/api-reference/sorted-set-collections) that allow you to access messages in many different ways. Consider the following use cases:
+
+* Paging messages N at a time
+* Retrieving messages in a certain time frame
+* Displaying messages in reverse order
+* Getting all messages ever sent
+
+Based on the parameters you use in your fetch command, you can easily satisfy any of the use cases above.
+
+A significant point to remember - **messages must be unique in sorted sets**! This means when saving the message to the sorted set, you should include some sort of message identifier or timestamp in your JSON.
+
+#### List
+
+A [list](../../cache/develop/basics/datatypes#lists) is a collection data type that stores an ordered array of elements. You can push elements to either the front or back of a list and you can add multiple entries at one time. A list is a great data type for instant messaging because it stores messages in the order they were sent. When users join an instant messaging session, you can quickly fetch the entire chat history with a single [listFetch API call](../../cache/develop/api-reference/list-collections#listfetch).
+
+Elements in a list do not need to be unique. You can push the same message into a list over and over again and it will be added every time. Lists are a good choice for instant messaging for the following use cases:
+
+* Paging messages N at a time
```

Comment:
nit: Can we say `Sending N messages` here?

Reply:
Sending? In what context? That word does not seem intuitive to me.